sdk/communication/Azure.Communication.Calling.Server/src/ConversationClient.cs

```diff
@@ -0,0 +1,546 @@
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="conversationId"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="callOptions"/> is null.</exception>
+        public virtual async Task<Response<JoinCallResponse>> JoinCallAsync(string conversationId, CommunicationIdentifier source, CreateCallOptions callOptions, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallClient)}.{nameof(JoinCallAsync)}");
+            scope.Start();
+            try
+            {
+                Argument.AssertNotNull(callOptions, nameof(callOptions));
+
+                JoinCallRequestInternal request = new JoinCallRequestInternal(
+                    CommunicationIdentifierSerializer.Serialize(source),
+                    callOptions.CallbackUri.AbsoluteUri,
+                    callOptions.RequestedModalities,
+                    callOptions.RequestedCallEvents);
+
+                return await RestClient.JoinCallAsync(
+                    conversationId,
+                    CommunicationIdentifierSerializer.Serialize(source),
+                    callOptions.CallbackUri.AbsoluteUri,
+                    callOptions.RequestedModalities,
+                    callOptions.RequestedCallEvents,
+                    null,
+                    cancellationToken
+                    ).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// Join the call using conversation id.
+        /// <param name="conversationId"> The conversation id that can be a group id or a encoded conversation url retrieve from client. </param>
+        /// <param name="source"> The source identity. </param>
+        /// <param name="callOptions"> The call Options. </param>
+        /// <param name="cancellationToken"> The cancellation token. </param>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="conversationId"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="callOptions"/> is null.</exception>
+        public virtual Response<JoinCallResponse> JoinCall(string conversationId, CommunicationIdentifier source, CreateCallOptions callOptions, CancellationToken cancellationToken = default)
```

Comment:
What does `source` mean here? Is not the call already created?

Reply:
the source is the the source identity which is going to join the call